### PR TITLE
Update to latest dhall to escape reserved keywords

### PR DESCRIPTION
**Pull request by TristanCacqueray**
This change updates the dhall-kubernetes-generator to use the latest dhall
version so that attribute named after reserved keywords are properly escape.
See: https://github.com/dhall-lang/dhall-haskell/issues/1896

### CODE REVIEW

**Comment by TristanCacqueray** [comment]
I guess using `allow-newer: true` is not ideal, but I'm not familiar enough with stack to figure a better solution.

**Comment by sjakobi** [comment]
> I guess using `allow-newer: true` is not ideal

I wouldn't worry about it. Ignoring a few version bounds is almost always unproblematic. If there is a problem, it usually shows as a build failure. And the dependency versions are still pinned.

**Comment by TristanCacqueray** [comment]
Note that the Kubernetes schemas are not affected by the issue this is fixing, and perhaps this could wait for the upcoming openapi-to-dhall utility.

**Comment by Gabriella439** [comment]
@TristanCacqueray: If you merge https://github.com/TristanCacqueray/dhall-kubernetes/pull/1 into your branch it should fix the Nix build and CI

File: dhall-kubernetes-generator/src/Main.hs [hunk 1]
```diff
@@ -11,6 +11,7 @@ import Data.Text (Text, pack)
 import Data.Void (Void)
 import Data.Yaml
 import Dhall.Core (Expr(..))
+import Dhall.Format (Format(..))
```
Comment (TristanCacqueray) [inline dhall-kubernetes-generator/src/Main.hs:14]: Note entirely sure why this is needed

**Comment by TristanCacqueray** [comment]
Thanks!

**Comment by Gabriella439** [comment]
@TristanCacqueray: Thank you! 🙂

**Pull request merged**